### PR TITLE
Drivers nrf52832 low power fixes

### PR DESCRIPTION
**Pull request by marc-cpdesign**

### CODE REVIEW

**Comment by teburd** [comment]
Thank you for your PR, I've added some nordic folks to take a look as they will have a better idea if this makes sense or not.

**Review by gmarull — COMMENTED** [review]
I think these fixes should go into nrfx. Otherwise, we'll end up with a high-level HAL + workarounds in Zephyr. If erratas need to go into Zephyr, let's switch to a low-level HAL or just simple register reads/writes, so we have clean drivers.

**Comment by marc-cpdesign** [comment]
ok, I can see the reasoning to put it in nrfx. If that's the way to go then, I guess that means I'll redo this for nrfx

**Comment by mbolivar-nordic** [comment]
> ok, I can see the reasoning to put it in nrfx. If that's the way to go then, I guess that means I'll redo this for nrfx

cc @nika-nordic

**Comment by nika-nordic** [comment]
> ok, I can see the reasoning to put it in nrfx. If that's the way to go then, I guess that means I'll redo this for nrfx

there is a plan to include workaround for this anomaly in nrfx 2.11, which will be probably delivered early next year. If you need it faster you can always open a PR to https://github.com/NordicSemiconductor/nrfx

**Comment by carlescufi** [comment]
@anangl @nika-nordic is this applicable still?

File: drivers/i2c/Kconfig.nrfx [hunk 1]
```diff
@@ -56,4 +56,14 @@ config I2C_3_NRF_TWIM
...
 
+config I2C_NRF_WORKAROUND_ERRATA89
+	bool "Workaround errata 89"
+	depends on SOC_NRF52832
+	help
+	  When GPIOTE (gpio interrupts) is active and SPI or TWI is
+	  also active, there is excessive current drain during low
+	  power (sleep/idle).
+	  This implements the workaround recommended in errata 89,
+	  which toggles the power to the TWI or SPI module after suspend.
```
Comment (anangl) [inline drivers/i2c/Kconfig.nrfx:67]: Anomaly 89 in nRF52832 is related to EasyDMA, so it concerns TWIM and SPIM, not TWI and SPI, which are a bit different peripherals.

**Review by anangl — CHANGES_REQUESTED** [review]

**Comment by anangl** [comment]
> @anangl @nika-nordic is this applicable still?

@carlescufi Not really. As @nika-nordic [mentioned above](https://github.com/zephyrproject-rtos/zephyr/pull/50076#issuecomment-1300146865), the workarounds for anomaly 89 will be implemented in nrfx drivers, so I think this PR should be closed.

File: drivers/i2c/Kconfig.nrfx (hunk 1 above)
Comment (marc-cpdesign) [reply drivers/i2c/Kconfig.nrfx:67]: Just to note,  although the errata does say it only applies to TWIM/SPIM, I had increased current drain regardless of whether SPI/TWI or SPIM/TWIM was used.

**Pull request closed without merging**